dialog/fileicon.go

```diff
@@ -0,0 +1,120 @@
+package dialog
+
+import (
+	"image/color"
+	"path/filepath"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+const (
+	fileIconSize = 64
+	fileTextSize = 24
+)
+
+type fileIcon struct {
+	widget.BaseWidget
+	parent  *fileDialog
+	current bool
+
+	path      string
+	icon      fyne.Resource
+	name, ext string
+	onTap     func(*fileIcon)
+}
+
+func (s *fileIcon) Tapped(_ *fyne.PointEvent) {
+	s.onTap(s)
+	s.current = true
+	s.Refresh()
+}
+
+func (s *fileIcon) TappedSecondary(_ *fyne.PointEvent) {
+
+}
+
+func (s *fileIcon) CreateRenderer() fyne.WidgetRenderer {
+	img := canvas.NewImageFromResource(s.icon)
+	text := widget.NewLabelWithStyle(s.name, fyne.TextAlignCenter, fyne.TextStyle{})
+	extText := canvas.NewText(s.ext, theme.BackgroundColor())
+	extText.Alignment = fyne.TextAlignCenter
+	extText.TextSize = theme.TextSize()
+	return &fileIconRenderer{icon: s,
+		img: img, text: text, ext: extText, objects: []fyne.CanvasObject{img, text, extText}}
+}
+
+func fileParts(path string) (name, ext string) {
+	name = filepath.Base(path)
+	ext = filepath.Ext(name[1:])
+	name = name[:len(name)-len(ext)]
+
+	if len(ext) > 1 {
+		ext = ext[1:]
+	}
+	return
+}
+
+func (f *fileDialog) newFileIcon(icon fyne.Resource, path string, tap func(*fileIcon)) *fileIcon {
+	name, ext := fileParts(path)
+	if icon == theme.FolderOpenIcon() {
+		name = "(Parent)"
+		ext = ""
+	}
+
+	ret := &fileIcon{
+		parent: f,
+		icon:   icon,
+		name:   name,
+		ext:    ext,
+		path:   path,
+		onTap:  tap,
+	}
+	ret.ExtendBaseWidget(ret)
+	return ret
+}
+
+type fileIconRenderer struct {
+	icon *fileIcon
+
+	ext     *canvas.Text
+	img     *canvas.Image
+	text    *widget.Label
+	objects []fyne.CanvasObject
+}
+
+func (s fileIconRenderer) Layout(size fyne.Size) {
+	iconAlign := (size.Width - fileIconSize) / 2
+	s.img.Resize(fyne.NewSize(fileIconSize, fileIconSize))
+	s.img.Move(fyne.NewPos(iconAlign, 0))
+	s.ext.Resize(fyne.NewSize(fileIconSize, fileTextSize))
+	ratioDown := 0.45
```

Comment:
hard-coded size (adjustment) outside a theme

Reply:
As this relates specifically to icons in the file dialog grid (currently) I don't think it's as heavily related to theme as the items above...